delfin/api/schemas/access_info.py

```diff
@@ -17,10 +17,36 @@
 update = {
     'type': 'object',
     'properties': {
-        'host': parameter_types.hostname_or_ip_address,
-        'port': parameter_types.tcp_udp_port,
-        'username': {'type': 'string', 'minLength': 1, 'maxLength': 255},
-        'password': {'type': 'string', 'minLength': 1, 'maxLength': 255},
+        'rest_access': {
+            'type': 'object',
+            'properties': {
+                'host': parameter_types.hostname_or_ip_address,
+                'port': parameter_types.tcp_udp_port,
+                'username': {'type': 'string', 'minLength': 1,
+                             'maxLength': 255},
+                'password': {'type': 'string', 'minLength': 1,
+                             'maxLength': 255}
+            },
+            'required': ['host', 'port', 'username', 'password'],
+            'additionalProperties': False
+        },
+        'ssh_access': {
+            'type': 'object',
+            'properties': {
+                'host': parameter_types.hostname_or_ip_address,
+                'port': parameter_types.tcp_udp_port,
+                'username': {'type': 'string', 'minLength': 1,
+                             'maxLength': 255},
+                'password': {'type': 'string', 'minLength': 1,
+                             'maxLength': 255},
+                'host_key': {'type': 'string', 'minLength': 1,
+                             'maxLength': 255},
+                'host_key_type': parameter_types.host_key_type
+            },
+            'required': ['host', 'port', 'username', 'password', 'host_key',
+                         'host_key_type'],
+            'additionalProperties': False
+        },
```

Comment:
If we sperate out rest and ssh as different options, I suggest to include SMI also as an option.. Because these three are most commonly used storage access mechanism..

Reply:
Hi @NajmudheenCT Thanks for you suggestion, If we support SMI-S, we need to consider it's related security issues together, I may need more investigation, so I suggest to do it in another PR, and I created an issue to trace it: https://github.com/sodafoundation/delfin/issues/252. Thanks.